src/mmw/js/src/analyze/views.js

```diff
@@ -406,12 +407,60 @@ var AnalyzeLayerToggleView = Marionette.ItemView.extend({
     },
 });
 
-var AnalyzeLayerToggleCollectionView = Marionette.CollectionView.extend({
-    childView: AnalyzeLayerToggleView,
+/**
+    A dropdown view for toggling multiple related layers
+    (eg the water quality tab has TN, TP, and TSS layers).
 
-    childViewOptions: function(model) {
+    Assumes only one layer of the list can be selected/active at a time. If multiple
+    can be selected, the active layername will appear as only the first active in
+    the list.
+**/
+var AnalyzeLayerToggleDropdownView = Marionette.ItemView.extend({
+    template: analyzeLayerToggleCollectionTmpl,
+
+    ui: {
+        'layers': 'a.layer-option',
+    },
+
+    events: {
+        'click @ui.layers': 'toggleLayer',
+    },
+
+    toggleLayer: function (e) {
+        e.preventDefault();
+        var code = $(e.target).data('id'),
+            layer = this.collection.findWhere({ code: code }),
+            layerTabCollection = App.getLayerTabCollection(),
+            layerGroup = layerTabCollection.findLayerGroup(layer.get('layerType'));
+        utils.toggleLayer(layer, App.getLeafletMap(), layerGroup);
+        layerTabCollection.trigger('toggle:layer');
+    },
+
+    initialize: function(options) {
+        this.collection = new Backbone.Collection(_.map(options.codes, function(code) {
+            var layer = App.getLayerTabCollection().findLayerWhere({ code: code });
+            if (layer) {
+                layer.on('change:active change:disabled', this.renderIfNotDestroyed, this);
+                return layer;
+            }
+        }, this));
```

Comment:
Good point. I was adapting this from `AnalyzeLayerToggleView` which didn't re-render properly with `modelEvents` because of the way I set the deferred pointsource layers up. Pushed a fixup to address.